### PR TITLE
CI check tree shaking

### PR DESCRIPTION
**Pull request by baldoalessandro**
Hello, here's my implementation of https://github.com/italia/design-react-kit/issues/733 using GitHub actions.
I still have some doubts about the implementation that, I think, are worth discussing:

- I've used an external action `unsplash/comment-on-pr@v1.3.0` is this OK or should I implement the functionality locally?
- The output of `agadoo` is quite long and doesn't fit the limit on comment length imposed by GitHub. One way to circumvent this problem would be to upload the result as an `artifact`(https://docs.github.com/en/actions/advanced-guides/storing-workflow-data-as-artifacts) and point the reader to it in the comment or, alternatively, split the result over multiple comments.

Let me know what you think, have a nice day!

Fixes #

#### PR Checklist

- [ ] My branch is up-to-date with the Upstream `master` branch.
- [ ] The unit tests pass locally with my changes (if applicable).
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable).
- [ ] I have added necessary documentation (if appropriate).

#### Short description of what this resolves:

This PR adds a CI job to check treeshake state of the library using GitHub actions.

#### Changes proposed in this Pull Request:

- set-up a GitHub action to run `agadoo` on pull requests
- post the result as a comment on the PR

### CODE REVIEW

**Comment by dej611** [comment]
I had some play with `agadoo` with this project and also for the #730 which has multiple chunks, but it always reports it as non-treeshakeable. Other than adding `sideEffect: false` I see no other optimizations for this package.

**Comment by dej611** [comment]
@baldoalessandro maybe we’re moving to github actions so there’s some hope to keep this PR as is.
Do you mind update the Pr with latest master?

**Comment by dej611** [comment]
Managed to make it pass the treeshake test with `agadoo` via #822 . @baldoalessandro are you still interested in working on this?

**Comment by baldoalessandro** [comment]
Hello, I was AFK for the holidays. I will rebase on master ASAP.

**Comment by dej611** [comment]
The post a comment to the PR seems to not work.

**Comment by baldoalessandro** [comment]
This is sadly expected as noted here: https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories

>With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository. **The permissions for the GITHUB_TOKEN in forked repositories is read-only**.  `[...]`

I think there isn't a practical way around this, but you can preview what the output will look like here on my fork https://github.com/baldoalessandro/design-react-kit/pull/1#issuecomment-1001698139

While putting together this answer I discovered that, while the GITHUB_TOKEN is invalidated at the end of the job, it is also possible to restrict the actual permissions granted by the token (see https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token). I can try to restrict the permissions granted to this job to the least privileges required for commenting the PR. Let me know if this is OK with you :)

File: .github/workflows/check-tree-shakeability.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          exit 0
+      - name: comment PR
+        uses: unsplash/comment-on-pr@v1.3.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          msg: |
+            **Check tree-shakeability** :deciduous_tree: for ${{ github.sha }}: ${{ steps.agadoo.outputs.status }}
+
+            **Agadoo stdout**
```
Comment (dej611) [inline .github/workflows/check-tree-shakeability.yml:75]: Let’s use a spoiler like thing here as the log can be very very long:

```md

<details>
  <summary>**Agadoo stout**</summary>
 …
</details>
```

And same thing for sterr

**Review by dej611 — CHANGES_REQUESTED** [review]

**Comment by dej611** [comment]
Good thinking about the token.
I’ve noticed the message can be too long for the output, I’ve put a suggestion to reduce it

**Review by dej611 — APPROVED** [review]

**Comment by dej611** [comment]
:tada: Thanks for your contribution @baldoalessandro !
PR approved

**Pull request merged**